src/kernels/MIOpenScatterND.cpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef MIOPEN_DONT_USE_HIP_RUNTIME_HEADERS
+#include <hip/hip_fp16.h>
+#include <hip/hip_runtime.h>
+#endif
+
+#include "float_types.h"
+#include "hip_atomic.hpp"
+#include "tensor_view.hpp"
+
+template <typename TIO, typename TINDEX>
+__device__ void ScatterNDAddForward_Kernel(const TIO* input,
+                                           const TINDEX* indices,
+                                           TIO* output,
+                                           size_t num_indices,
+                                           size_t slice_size,
+                                           size_t slice_dim,
+                                           tensor_view_t<9> shape)
+{
+    size_t gid = blockIdx.x * blockDim.x + threadIdx.x;
+
+    size_t i           = 0;
+    bool out_of_bounds = false;
+    size_t indices_idx = gid / slice_size;
+    size_t slice_idx   = gid % slice_size;
+
+    if(indices_idx >= num_indices)
+        return;
+
+    size_t output_shape_prefix[9];
+    size_t batch_strides[9];
+
+    for(int j = 0; j < slice_dim; j++)
```

Comment:
```suggestion
    for(size_t j = 0; j < slice_dim; j++)
```
comparation between int and size_t may lead to fail in CI/CD

Reply:
updated.